[CUBRIDMAN-266] [EN] 변경된 PL 서버 (cub_pl) 동작 변경에 대한 설명 추가/변경 (#624)
CUBRID/cubrid-manual: en/env.rst

    
        *   cub_master: **/tmp**
        *   cub_broker process: **$CUBRID/var/CUBRID_SOCK**
        *   cub_pl process: **$CUBRID/var/CUBRID_SOCK**

.. note::

>     * **java.io.tmpdir** among **stored_procedure_vm_options** is an option that specifies the directory where temporary files of the cub_pl process (built-in Java VM) are created. If the **CUBRID_TMP** environment variable is set, the value of **java.io.tmpdir** is ignored, and the directory specified by CUBRID_TMP will be used instead.

Reviewer: ```suggestion
    * The **java.io.tmpdir** option within the **stored_procedure_vm_options** specifies the directory where temporary files for the cub_pl process (the built-in Java VM) are created. If the **CUBRID_TMP** environment variable is set, the value of **java.io.tmpdir** is ignored, and the directory specified by CUBRID_TMP will be used instead.
```

>     * **java.io.tmpdir** among **cub_plstored_procedure_jvm_options** is an option that specifies the directory where temporary files of the cub_pl process (built-in Java VM) are created. If the **CUBRID_TMP** environment variable is set, the value of **java.io.tmpdir** is ignored, and the directory specified by CUBRID_TMP will be used instead.

Reviewer 2: ```suggestion
    * **java.io.tmpdir** among **stored_procedure_vm_options** is an option that specifies the directory where temporary files of the cub_pl process (built-in Java VM) are created. If the **CUBRID_TMP** environment variable is set, the value of **java.io.tmpdir** is ignored, and the directory specified by CUBRID_TMP will be used instead.
```

*   Path: In the Windows environment, the **%CUBRID%\\bin** and **%CUBRID%\\cci\\bin**, which is a directory that contains CUBRID system's execution file, must be included in the **Path** environment variable.

> *   JAVA_HOME: To use the Stored procedure in the CUBRID system, the Java Virtual Machine (JVM) version 1.8 or later must be installed, and the **JAVA_HOME** environment variable must designate the concerned directory. See the :ref:`cubrid-pl-server-config`.

Reviewer 2: jdk8 내장에 따라 해당 환경 변수도 변경이 필요합니다.

> *   JAVA_HOME: To use the Stored procedure in the CUBRID system, the Java Virtual Machine (JVM) version 1.8 or later must be installed, and the **JAVA_HOME** environment variable must designate the concerned directory. See the :ref:`cubrid-pl-server-config`.

Reviewer: ```suggestion
*   JAVA_HOME: To use stored procedures in the CUBRID system, the Java Virtual Machine (JVM) version 1.8 or later must be installed, and the **JAVA_HOME** environment variable must designate the concerned directory. See the :ref:`cubrid-pl-server-config`.
```

> *   JVM_PATH: To use the Stored procedure in the CUBRD system, the **JVM_PATH** environment variable can specify the JVM library (libjvm) path explicitly instead of finding the library from **JAVA_HOME**. See the :ref:`cubrid-pl-server-config`.

Reviewer 2: jdk8 내장에 따라 해당 환경 변수도 변경이 필요합니다.

> *   JVM_PATH: To use the Stored procedure in the CUBRD system, the **JVM_PATH** environment variable can specify the JVM library (libjvm) path explicitly instead of finding the library from **JAVA_HOME**. See the :ref:`cubrid-pl-server-config`.

Reviewer: ```suggestion
*   JVM_PATH: To use stored procedures in the CUBRD system, the **JVM_PATH** environment variable can specify the JVM library (libjvm) path explicitly instead of finding the library from **JAVA_HOME**. See the :ref:`cubrid-pl-server-config`.
```

*   The port used when the CUBRID Manager client accesses the CUBRID Manager server process is **cm_port** of the cm.conf. The default value is 8001.

Ports for CUBRID Procdedure Language (PL) Server
---------------------------------------------

> The following table summarizes the ports, based on the listening processes, used for CUBRID Procdedure Language (PL) server. The ports are identical regardless of the OS type.

Reviewer 3: Procdedure 오타 입니다.

Author: ```suggestion
The following table summarizes the ports, based on the listening processes, used for CUBRID Procedural Language (PL) server. The ports are identical regardless of the OS type.
```

+---------------+--------------+----------------------------+--------------------------+
| Listener      | Requester    | Port                       | Firewall Port Setting    |
+===============+==============+============================+==========================+
| cub_pl        | cub_server   | stored_procedure_port      | Open                     |
+---------------+--------------+----------------------------+--------------------------+

Behavior according to the OS type and the **stored_procedure_uds** setting in **cubrid.conf**

*   Linux
  - If **stored_procedure_uds** is set to **yes**, a UNIX domain socket is used and a TCP socket is not used.
  - If **stored_procedure_uds** is set to **no**, a TCP socket is used, and the port is determined by the **stored_procedure_port** setting.

*   Windows
  - Windows does not support UNIX domain sockets.
  - Therefore, regardless of the **stored_procedure_uds** setting, a TCP socket is used, and the port is determined by the **stored_procedure_port** setting.

.. note::

>         The port used when the CUBRID Procedure Language server (cub_pl) communicates with the cub_server is specified by **stored_procedure_port** in **cubrid.conf**. The default value is 0, which means a random available port is assigned.

Reviewer: ```suggestion
        The port used for communication between the CUBRID Procedure Language server (cub_pl) and cub_server is specified by **stored_procedure_port** in **cubrid.conf**. The default value is 0, which means that a random available port will be assigned.
```
